avocado/utils/partition.py

```diff
@@ -50,30 +50,29 @@ def __str__(self):
                                       super(PartitionError, self).__str__())
 
 
-class MtabLock(object):
+class MtabLock(filelock.FileLock):
     mtab = None
+    device = "/etc/mtab"
+
+    def __init__(self, timeout=60):
+        device_hash = hashlib.sha1(self.device.encode("utf-8")).hexdigest()
+        lock_filename = os.path.join(tempfile.gettempdir(), device_hash)
+        super(MtabLock, self).__init__(lock_filename, timeout=timeout)
 
     def __enter__(self):
-        self.mtab = open("/etc/mtab")
-        end_time = time.time() + 60
-        while time.time() < end_time:
-            try:
-                fcntl.flock(self.mtab.fileno(),
-                            fcntl.LOCK_EX | fcntl.LOCK_NB)
-                break
-            except IOError as details:
-                if details.errno == 11:
-                    time.sleep(0.1)
-                else:
-                    raise
+        try:
+            super(MtabLock, self).__enter__()
+        except (filelock.LockFailed, filelock.AlreadyLocked) as e:
+            reason = "Unable to obtain '/etc/mtab' lock in %ds" % self.timeout
```

Comment:
Probaby you should also get the device from the new created attribute `self.device` here for the message. For consistency.

Besides that, I think you forgot to put the `timeout` init parameter in a instance attribute (self). As none test got this, it could be good see if it's not necessary to write a proper test.